### PR TITLE
NEW: New supervisor (IPSet)

### PR DESCRIPTION
**Pull request by bvandewalle**
An Alternative implementation of the Supervisor by using IPSets. It should reduce complexity by an order of magnitude.

### CODE REVIEW

File: monitor/docker.go [hunk 1]
```diff
@@ -334,23 +324,15 @@ func (d *dockerMonitor) syncContainers() error {
 
 		if err != nil {
 			log.WithFields(log.Fields{
-				"package":   "monitor",
-				"monitor":   d,
-				"error":     err,
-				"container": container,
+				"package": "monitor",
```
Comment (satyamsi) [inline monitor/docker.go]: Shouldn't we have the container Ids which failed in the log ?

File: monitor/docker.go [hunk 2]
```diff
@@ -334,23 +324,15 @@ func (d *dockerMonitor) syncContainers() error {
...
+				"error":   err,
 			}).Error("Error Syncing existing Container")
-
```
Comment (satyamsi) [inline monitor/docker.go]: How do we deal with errors while syncing containers ?

File: monitor/docker.go (hunk 1 above)
Comment (bvandewalle) [inline monitor/docker.go]: Yes, but the issue is that in some cases the "Container" object could be nil. (not given by the Docker API).

File: monitor/docker.go (hunk 2 above)
Comment (bvandewalle) [inline monitor/docker.go]: When syncing, if an error happens (Example the PolicyResolver fails), we simply do not activate the container in Trireme. Depending on the default policy, this means that the container will/will not be the open Network Connections

File: monitor/docker.go (hunk 1 above)
Comment (bvandewalle) [inline monitor/docker.go]: We were actually logging the container.id in the past, but that could result in a Kernel panic if the container object is nil. Ideally this should never happen and be caught by the Docker event APi upstream ...

Comment (satyamsi) [inline monitor/docker.go]: so it will print nil but in other cases we will see the id.

File: monitor/docker.go (hunk 2 above)
Comment (satyamsi) [inline monitor/docker.go]: ok

**Pull request merged**